sdk/eventgrid/azure-messaging-eventgrid/swagger/src/main/java/EventGridCustomization.java

```diff
@@ -229,63 +236,86 @@ public void customizeResourceEvents(LibraryCustomization customization, Logger l
-                    .getJavadoc()
-                    .setDescription("Set the authorization property: The requested authorization for the operation.")
-                    .setParam("authorization", "the authorization value to set.")
-                    .setReturn(String.format("the %s object itself.", className))
-                    .setDeprecated(String.format("This method is no longer supported since v4.9.0. <p> Use {@link %s#setResourceAuthorization(ResourceAuthorization)} instead.", className));
-
-                classCustomization.getMethod("getHttpRequest")
-                    .getJavadoc()
-                    .setDescription("Get the httpRequest property: The details of the operation.")
-                    .setReturn("the httpRequest value.")
-                    .setDeprecated(String.format("This method is no longer supported since v4.9.0. <p> Use {@link %s#getResourceHttpRequest()} instead.", className));
-
-                classCustomization.getMethod("setHttpRequest")
-                    .getJavadoc()
-                    .setDescription("Set the httpRequest property: The details of the operation.")
-                    .setParam("httpRequest", "the httpRequest value to set.")
-                    .setReturn(String.format("the %s object itself.", className))
-                    .setDeprecated(String.format("This method is no longer supported since v4.9.0. <p> Use {@link %s#setResourceHttpRequest(ResourceHttpRequest)} instead.", className));
 
+                classCustomization.customizeAst(compilationUnit -> {
+
+                    ClassOrInterfaceDeclaration classDec = compilationUnit.getClassByName(className).get();
+                    compilationUnit.addImport("com.azure.core.util.logging.ClientLogger");
+                    compilationUnit.addImport("com.azure.core.util.serializer.JacksonAdapter");
+                    compilationUnit.addImport("com.azure.core.util.serializer.SerializerAdapter");
+                    compilationUnit.addImport("com.azure.core.util.serializer.SerializerEncoding");
+                    compilationUnit.addImport("java.io.IOException");
+                    compilationUnit.addImport("java.io.UncheckedIOException");
+                    classDec.addFieldWithInitializer("ClientLogger", "LOGGER", parseExpression("new ClientLogger(" + className + ".class)"), Keyword.STATIC, Keyword.FINAL, Keyword.PRIVATE);
+                    classDec.addFieldWithInitializer("SerializerAdapter", "DEFAULT_SERIALIZER_ADAPTER", parseExpression("JacksonAdapter.createDefaultSerializerAdapter()"), Keyword.STATIC, Keyword.FINAL, Keyword.PRIVATE);
+
+
+                    Arrays.asList("Authorization", "Claims", "HttpRequest").forEach(method -> {
+                        classDec.getMethodsByName("get" + method).get(0).setName("getResource" + method);
+                        classDec.getMethodsByName("set" + method).get(0).setName("setResource" + method);
```

Comment:
May want to use a slightly safer design like

```java
classDec.getMethodByName("get" + method).ifPresent(getter -> getter.setName("getResource" + method));
classDec.getMethodByName("set" + method).ifPresent(setter -> setter.setName("setResource" + method));
```

Reply:
There is no `getMethodByName` in `classDec`. This returns a list and we should ensure all methods contained in that list get renamed as we may add overloads. So, this should be more like:

`classDec.getMethodsByName("oldName").forEach(methodDeclaration -> methodDeclaration.setName("newName")`

Reply:
done! good call.